papyri/gen.py

```diff
@@ -1191,146 +1286,68 @@ def get_example_data(
-                                    )
-                                plt.close("all")
-                            fig_managers = executor.fig_man()
-                            assert len(fig_managers) == 0, fig_managers + [
-                                did_except,
-                            ]
-                    # we've executed, we now want to compare output
-                    # in the docstring with the one we produced.
-                    if (out == repr(res)) or (res is None and out == []):
-                        pass
-                    else:
-                        pass
-                        # captured output differ TBD
-                entries = parse_script(script, ns=ns, prev=acc, config=config, where=qa)
-                if entries is None:
-                    entries = [("jedi failed", "jedi failed")]
-                entries = _add_classes(entries)
-                tok_entries = [GenToken(*x) for x in entries]  # type: ignore
 
-                acc += "\n" + script
-                example_section_data.append(
-                    Code(tok_entries, "\n".join(item.out), ce_status)
-                )
-                for figname, _ in figs:
-                    example_section_data.append(
-                        Fig(
-                            RefInfo.from_untrusted(
-                                self.root, self.version, "assets", figname
-                            )
-                        )
-                    )
-                all_figs.extend(figs)
+        sys_stdout = sys.stdout
+
+        def dbg(*args):
+            for arg in args:
+                sys_stdout.write(f"{arg}\n")
+            sys_stdout.flush()
+
+        try:
+            filename = inspect.getfile(obj)
+        except TypeError:
+            filename = None
+        try:
+            lineno = inspect.getsourcelines(obj)[1]
+        except (TypeError, OSError):
+            lineno = None
+
+        doctest_runner = PapyriDocTestRunner(
+            gen=self,
+            obj=obj,
+            qa=qa,
+            config=config,
+            # TODO: Make optionflags configurable
+            optionflags=doctest.ELLIPSIS,
+        )
+        example_section_data = Section([], None)
+
+        def debugprint(*args):
+            """
+            version of print that capture current stdout to use during testing to debug
+            """
+            sys_stdout.write(" ".join(str(x) for x in args) + "\n")
+
+        blocks = doctest.DocTestParser().parse(example_code, name=qa)
+        for block in blocks:
+            if isinstance(block, doctest.Example):
+                doctests = doctest.DocTest([block],
+                                           globs=doctest_runner.globs,
+                                           name=qa, filename=filename,
+                                           lineno=lineno, docstring=example_code)
+                if exec:
```

Comment:
```suggestion
                if config.exec:
```

Otherwise that's the `exec` function which will always be true.